portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,25 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.UpdateRequest
+import cromwell.pipeline.datastorage.dto.{User, UserId}
+import cromwell.pipeline.utils.StringUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Random
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+
+  def update(id: String, request: UpdateRequest): Future[Int] = {
+    val updatedUser = User(
+      userId = UserId(id),
+      email = request.email,
+      passwordSalt = Random.nextLong().toHexString,
+      passwordHash = StringUtils.calculatePasswordHash(request.password, Random.nextLong().toHexString),
```

Comment:
The password salt which is used for generating a password hash should be the same as used for the `passwordSalt` field.
Otherwise, your user won't log in because the password hash generated during the validation process and stored in the database will be different.

Reply:
done